vcd/datasource_vcd_vapp_test.go

```diff
@@ -0,0 +1,114 @@
+// +build vapp ALL functional
+
+package vcd
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/vmware/go-vcloud-director/v2/govcd"
+)
+
+// getAvailableVapp collects one available Vapp to use in data source tests
+func getAvailableVapp() (*govcd.VApp, error) {
```

Comment:
What happens here? why we don't use setup prepared `firstVapp` ?

Reply:
findFirstVapp is in go-vcloud-director.

Reply:
Does it means that this test expects that other tests create and left vApp ?

Reply:
No. It means that if we find a vApp, we test the data source. If not, we skip the test